### PR TITLE
[spirv] Update doc and SPIRV-Tools

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: docs/SPIR-V.rst [hunk 1]
```diff
@@ -286,6 +286,58 @@ interface variables:
...
+* ``-fcgl``: turn off legalization and optimization
+* ``-Od``: turn off optimization
+* ``-Vd``: turn off validation
+
+Legalization
+~~~~~~~~~~~~
+
+HLSL is a fairly permissive language considering the flexibility it provides for
+manipulating resource objects. The developer can create local copies, pass
+around them as function parameters and return values, as long as after certain
```
Comment (ehsannas) [inline docs/SPIR-V.rst]: s/pass around them/pass them around

File: docs/SPIR-V.rst [hunk 2]
```diff
@@ -286,6 +286,58 @@ interface variables:
...
+transformations (function inlining, constant evaluation and propagating, dead
+code elimination, etc.), the compiler can remove all temporary copies and
+pinpoint all uses to unique global resource objects.
+
+Resulting from the above property of HLSL, if we translate into SPIR-V for
+Vulkan literally from the input HLSL source code, we will sometimes generate
+illegal SPIR-V. Certain transformations are needed to legalize the literally
+translated SPIR-V. Performing such transformations at the frontend AST level
+is cumbersome or impossible (e.g., function inlining). They are better to be
+conducted at SPIR-V level. Therefore, legalization are delegated to SPIRV-Tools.
```
Comment (ehsannas) [inline docs/SPIR-V.rst]: s/are/is

File: docs/SPIR-V.rst (hunk 1 above)
Comment (antiagainst) [reply docs/SPIR-V.rst]: Done

File: docs/SPIR-V.rst (hunk 2 above)
Comment (antiagainst) [reply docs/SPIR-V.rst]: Done

**Comment by antiagainst** [comment]
Only .rst change. Previously passed buildbot.

**Pull request merged**